src/components/tasks/card/index.tsx

```diff
@@ -210,6 +213,78 @@ const Card: FC<Props> = ({
     actions.onEditRoute()
   }
 
+  const hasIssueAssignee = () => cardDetails.github?.issue.assignee ?? false;
+  const hasTaskAssignee = () => cardDetails.assignee ?? false;
+  const isIssueClosed = () => cardDetails.github?.issue?.status === "closed";
+  const isTaskComplete = () => cardDetails.status === "Completed";
+
+  const showAssignButton = () =>
+		hasIssueAssignee() && !hasTaskAssignee() && !isIssueClosed() && !isTaskComplete();
+
+  // assign the task to the issue assignee
+  const handleAssignToIssueAssignee = async () => {
+		setIsLoading(true);
+		try {
+      const data: updateTaskDetails = {
+        assignee: cardDetails.github?.issue.assigneeRdsInfo?.username,
+        status: "ASSIGNED",
+      }
+      
+      // Update start date when assigning the task to the issue assignee
+      if(!cardDetails.startedOn) {
+        data.startedOn = new Date().getTime() / 1000;
+      }
+
+			const { requestPromise } = fetch({
+				url: `${TASKS_URL}/${cardDetails.id}`,
+				method: "patch",
+				data,
+			});
+			await requestPromise;
+   
+      updateTask(cardDetails.id, data);
+			toast(SUCCESS, "Task assigned successfully!");
+			setIsLoading(false);
+		} catch (err: any) {
+			setIsLoading(false);
+			if ("response" in err) {
+				toast(ERROR, err.response.data.message);
+				return;
+			}
+			toast(ERROR, err.message);
+		}
+	};
+
+  const getFormattedClosedAtDate = () => {
+    const closedAt = cardDetails?.github?.issue?.closedAt;
+    return getDateInString(new Date(closedAt ?? ""));
```

Comment:
`new Date("")` gives `invalid Date`

Reply:
Actually `closedAt` will never be null. `getFormattedClosedAtDate()` is only invoked when the issue has already been closed. There's a check added for this. That's why I added `""`.

But, I've anyway replaced `""` with `Date.now()`.